addons/portainer/enable

```diff
@@ -16,6 +16,18 @@ HELM = os.path.expandvars("$SNAP/microk8s-helm3.wrapper")
 MICROK8S_STATUS = os.path.expandvars("$SNAP/microk8s-status.wrapper")
 MICROK8S_ENABLE = os.path.expandvars("$SNAP/microk8s-enable.wrapper")
 
+def mark_portainer_enabled():
+  """
+  Mark the portainer addon as enabled by creating portainer.enabled
+  """
+  try:
+      snapdata_path = os.environ.get("SNAP_DATA")
+      lock_fname = "{}/var/lock/portainer.enabled".format(snapdata_path)
+      subprocess.call(['sudo', 'touch', lock_fname])
```

Comment:
The use of this sudo here may be problematic. Maybe you need a message letting the use know why the sudo password is needed?

Reply:
Firstly, thank you for taking a look at the PR and for the testing tip! I've asked a team mate to have a review tomorrow. Once it looks good to them, I'll let you know that it's ready for another review. 

It seems that with a default install of MicroK8s, the SNAP_DATA path is write protected for non-root users, so sudo is required? 
![image](https://github.com/canonical/microk8s-community-addons/assets/83188384/2e6831d4-991f-43de-b147-6b36946c3ade)

I noticed that a few other community addons, like Kata put their file for `check_status` in the SNAP_DATA path. Is it best practice to put it there?

Reply:
Using this kind of lock/state-files is not a good practice. This lock files are present only on the node where the addon is enabled from so if you have a multinode cluster `microk8s status` on other nodes will report the addon as disabled. It would be best to use a resource from k8s (eg a pod or a service) to mark an addon as enabled/disabled.

Reply:
Hey @ktsakalozos! I was asking the team about updating the helm chart and they were mentioning that changing the Portainer Helm Chart and check_status could cause issues. For example:
 
Given I change the Portainer addon check_status to `pod/portainer-server` 
And I update the Portainer Helm Chart to have a pod name `pod/portainer-server...`,  
When a user runs a new MicroK8s version with the new check_status
And they already have an old instance of Portainer Server running 
Then the Portainer addon won't be detected as enabled

As a solution, would it be possible to update [the `is_enabled` util function](https://github.com/canonical/microk8s/blob/40275f34e2ae1dc74b5fed052d427c43ac2a2302/scripts/wrappers/common/utils.py#L553-L579) for MicroK8s to allow a regex check if the check status contains a second term, 'regex'? 

The change to the is_enabled util function could be something like the following:
```py
import re

def is_enabled(addon, item):
    # a filepath could contain ' regex\', so we need to split it
    addonTerms = addon.split(" ")
    if len(addonTerms) > 1 and addonTerms[1] == "regex":
        try:
            if re.search(addonTerms[0], item):
                return True
        except re.error:
            pass
    elif addon in item:
        return True
    else:
        filepath = os.path.expandvars(addon)
        return os.path.isfile(filepath)
```

I'm happy to make a small PR to both the community addons repo (to use a regex term for the portainer check_status, and update the documentation comment around check_status) and the MicroK8s repo (to update the is_enabled function).

Would this be ok?

Reply:
How would the regex be in this case? There should be some way to figure out the version. I suspect a regex will not be enough.

Reply:
Ah - good point about the old versions of MicrokK8s needing to use the check status from the up to date community addons repo. I mistakenly assumed that old versions of MicroK8s would use an old version of this repo. 

Reusing the `check_status` key might not be good, since the regex I'd like to use for Portainer is `"^pod\/portainer(?!-agent).*$"` so that `pod/portainer<any following text>` will pass, but `pod/portainer-agent<any following text>` will fail.

Instead of replacing the `check_status`, could we keep it (for old MicroK8s versions to use as is) and have the new field in `addons.yaml` be `regex_check_status` without needing `use_regex`? 

If the `regex_check_status` exists, then the new MicroK8s version should search for it, otherwise MicroK8s can use the existing search method. The change to the is_enabled function could be something like:

```py
def is_enabled(addon, row):
    check_status = addon["check_status"]
    if "regex_check_status" in addon and addon["regex_check_status"]:
        regex_check_status = addon["regex_check_status"]
        return re.search(regex_check_status, row)
    elif check_status in row:
        return True
    else:
        filepath = os.path.expandvars(check_status)
        return os.path.isfile(filepath)
```

Is there a case where we need to figure out the version? I might still be missing something.

Reply:
> I mistakenly assumed that old versions of MicroK8s would use an old version of this repo.

Old versions of MicroK8s will indeed use an the respective versions on the addons found in the matching branches.


> Instead of replacing the `check_status`, could we keep it (for old MicroK8s versions to use as is) and have the new field in `addons.yaml` be `regex_check_status` without needing `use_regex`?

Yes, it is best we do not overload the `check_status` field.


> Is there a case where we need to figure out the version?

Just note that the "old" addons repository will not have this functionality. Any changes we do will go into the 1.29 release.

Looking forward to seeing your PRs. Thank you.